src/validationinterface.h

```diff
@@ -98,6 +98,17 @@ class CValidationInterface {
      * Called on a background thread.
      */
     virtual void TransactionAddedToMempool(const CTransactionRef& tx) {}
+    /**
+     * Notifies listeners of a transaction having been added to mempool.
+     *
+     * Required because of legacy ZMQ notifications including block
+     * transactions when using TransactionAddedToMempool. See:
+     * CZMQNotificationInterface::TransactionAddedToMempool
```

Comment:
Ugly hack. Just change CZMQNotificationInterface

Reply:
Sorry, change it how specifically? If you mean I should change the legacy notifications too, I disagree? Might be misunderstanding.

Reply:
Figured out what you meant. Done in follow-on commit.